vsintegration/src/FSharp.ProjectSystem.Base/Project/ProjectFactory.cs

```diff
@@ -112,6 +112,63 @@ protected override void CreateProject(string fileName, string location, string n
                 }
             }
 
+            // Solution properties
+            IVsSolution solution = this.Site.GetService(typeof(SVsSolution)) as IVsSolution;
+            Debug.Assert(solution != null, "Could not retrieve the solution service from the global service provider");
+
+            // We do not want to throw. If we cannot set the solution related constants we set them to empty string.
+            string solutionDirectory, solutionFile, userOptionsFile;
+            solution.GetSolutionInfo(out solutionDirectory, out solutionFile, out userOptionsFile);
+            if (solutionDirectory == null)
+            {
+                solutionDirectory = String.Empty;
+            }
+            if (solutionFile == null)
+            {
+                solutionFile = String.Empty;
+            }
+            string solutionFileName = (solutionFile.Length == 0) ? String.Empty : Path.GetFileName(solutionFile);
+            string solutionName = (solutionFile.Length == 0) ? String.Empty : Path.GetFileNameWithoutExtension(solutionFile);
+            string solutionExtension = String.Empty;
+            if (solutionFile.Length > 0 && Path.HasExtension(solutionFile))
+            {
+                solutionExtension = Path.GetExtension(solutionFile);
+            }
+
+            // DevEnvDir property
+            IVsShell shell = this.Site.GetService(typeof(SVsShell)) as IVsShell;
+            Debug.Assert(shell != null, "Could not retrieve the IVsShell service from the global service provider");
+            object installDirAsObject;
+
+            // We do not want to throw. If we cannot set the solution related constants we set them to empty string.
+            shell.GetProperty((int)__VSSPROPID.VSSPROPID_InstallDirectory, out installDirAsObject);
+            string installDir = ((string)installDirAsObject);
+            if (String.IsNullOrEmpty(installDir))
+            {
+                installDir = String.Empty;
+            }
+            else 
+            {
+                // Ensure that we have trailing backslash as this is done for the langproj macros too.
+                if (installDir[installDir.Length - 1] != Path.DirectorySeparatorChar)
+                {
+                    installDir += Path.DirectorySeparatorChar;
+                }
+            }
+
+            var projectGlobalPropertiesThatAllProjectSystemsMustSet = new Dictionary<string, string>()
+            {
+                { GlobalProperty.SolutionDir.ToString(), solutionDirectory },
+                { GlobalProperty.SolutionPath.ToString(), solutionFile },
```

Comment:
Suggestion: rename `solutionFile` to `solutionPath` to match the property being set.